solr/contrib/prometheus-exporter/src/java/org/apache/solr/prometheus/scraper/SolrCloudScraper.java

```diff
@@ -135,12 +135,9 @@ public MetricSamples collections(MetricsQuery metricsQuery) throws IOException {
     return request(solrClient, metricsQuery);
   }
 
-  private Set<String> getBaseUrls() throws IOException {
-    return solrClient.getClusterStateProvider().getClusterState().getCollectionsMap().values()
-        .stream()
-        .map(DocCollection::getReplicas)
-        .flatMap(List::stream)
-        .map(Replica::getBaseUrl)
+  private Set<String> getBaseUrls() {
+    return solrClient.getClusterStateProvider().getLiveNodes().stream()
```

Comment:
might this include overseer nodes that don't have any data on them? maybe that's ok

Reply:
It would do, but thats fine too be honest, this just allows us to collect metrics from those nodes